### DISCUSSION TITLE
Question: Order routing for international / localised fullfilment

### DISCUSSION THREAD

**Discussion by Krislunde** [swellstores/community#510] [Help] [answered]
We are currently pitching Swell as the e-commerce solution for a new client.

The client will be expanding to more countries in the near future, and will establish local fulfilment/warehouses in the countries they expand to.

Swell documentation helpfully says that it is possible to enable [multiple locations](https://www.swell.is/help/store-setup/shipping#enabling-multiple-locations) which is useful for calculating shipping from these locations. However, it doesn't mention whether it is possible to track stock levels on an individual location basis. I am assuming this is possible based on [physical products](https://developers.swell.is/guides/core-concepts/shipping-and-fulfillment#physical-products) mentioning that "Swell supports multiple stock locations" and [real time rates](https://developers.swell.is/guides/core-concepts/shipping-and-fulfillment#real-time-rates) saying that it takes available stock into consideration. 

**Question 1)** Please confirm whether Swell supports individual stock tracking in multiple locations?

The client has also put down the requirement that they need order forwarding. So if an orders delivery address is in germany, the order should be forwarded to the german fulfilment warehouse - and they can only see the orders that pertain to them. Under [shipping and fulfilment](https://developers.swell.is/guides/core-concepts/shipping-and-fulfillment#shipping-and-fulfillment) Swell says that multiple locations are supported, but mentions nothing about fulfilment. Neither does [multiple locations](https://www.swell.is/help/store-setup/shipping/shipping-zones#multiple-locations).

**Question 2)** Is it possible to do order forwarding/routing? We thought maybe we could suggest that as a fallback for now,  each fulfilment location would have to use the Swell dashboard, but it's not even possible to filter/sort orders by Country of delivery address - making it a hassle to individually check each single unfulfilled order to see if any are relevant to your country. You can't even add more fields to sort by. Only rearrange or hide them.
![image](https://github.com/swellstores/community/assets/5624240/7dafd731-18d8-48d0-a698-e95de15ba321) 
There also doesn't seem to be any way of adding users with granular access rights or similar.

Am I correct in assuming that frontends for each fulfilment center would have to be developed, doing something like fetching orders via the backend api, only show the ones relevant to the fulfilment center in question by filtering on the country code in the shipping section of the order objects, and developing order fulfilment functionality to this new frontend? That's a lot of extra work essentially just duplicating the Swell dashboard.

Do you have any other clever suggestions?

**Comment by sags95** [comment, marked as answer]
Hi @Krislunde, currently by default we do not support multiple inventory stock tracking. There is a `location` field on the stock model, but we don't sum stock values per location: https://developers.swell.is/backend-api/stock. 

However we do have many merchants that have built out custom inventory models (or customized the default one) by using custom fields or a custom model to account for multiple inventory locations.

For your second question, this would be possible with a new feature called Views, this is currently in developer preview but I'll be happy to share documentation with you and provide access. Essentially this will let you define new content on the Admin dashboard and interact with records. Views can be placed on standard models (such as products, orders, etc) or at the top-level on the dashboard. Will send you details in a DM for access.

**Comment by Krislunde** [comment]
Hi @sags95, thanks for your reply!

>However we do have many merchants that have built out custom inventory models (or customized the default one) by using custom fields or a custom model to account for multiple inventory locations.

What's the intention of the `location` field on the stock model, if there's no multi-location tracking? Could you please elaborate on how this can be achieved with custom fields or custom models? Some logic would also have to be created to properly remove stock items with a location corresponding to the area where a purchase was made/fulfilled, no? Like, we might have to show "no stock available" to customers in Germany if Germany fulfilment doesn't have any more stock. I'm assuming this wouldn't be possible? How do other merchants deal with higher shipping costs if having to ship from more remote warehouses? Splitting orders between warehouses also sounds like a potential headache?

Super interested in hearing more about how other merchants have solved this issue.

PS. In order to foster a great community, I think Swell should encourage users who create custom solutions like this to write blog posts about it and publish it on the swell site. [Strapi](https://strapi.io/blog) for example does a really good job of this, with articles written both by staff and community members.

**Reply by ericingram** [reply]
@Krislunde The field is there because some users have implemented their own stock allocation strategies, and we intend to eventually enhance the model to support it better natively. We may implement a full native solution in the future, but that hasn't been decided and likely wouldn't be a priority in the near future. We'd love to see Swell App developers comes up with solutions as well.

Currently it would require setting up a separate model and service, and custom checkout depending on the desired behavior. Once we introduce App Event Hooks it will be possible to make an app do it while integrating with all standard functionality including hosted checkout.

Internally we're very focused getting Swell Apps launched so that there doesn't need to be as much custom work involved in things like this, which should also be a great boost for the community!
